### DISCUSSION TITLE
Suricata suppression & Rule Categories

### DISCUSSION THREAD

**Discussion by 8inaryMata1eao** [Security-Onion-Solutions/securityonion#8311] [Unsupported Versions] [answered]
A frequent alert being triggered by Suricata is the "ET TOR Known Tor Relay" or "ET TOR Known Tor Exit".  This alert is for a public facing asset which provides no value when triggered.  My question is, while I know I can individually create suppression rules for each SID and the public hosts, this does not scale.  Does anyone know of a method to create suppression rules to do something like using the so-rule modify but target rule categories with the strings "ET TOR Known" and use a variable $PUBLIC_SERVERS or something? The same way the disabled works with "re:ET TOR Known".  
Thanks

![image](https://user-images.githubusercontent.com/80425093/179281234-abdc01e4-33e9-4bd9-9347-6c29bf032ff4.png)

**Comment by btdewey** [comment]
You can disable the rules based on a regular expression per the security onion tuning guide.  It's more complicated, but possible.
https://docs.securityonion.net/en/2.3/managing-alerts.html
If you want to disable multiple rules at one time, you can use a regular expression, but make sure you enclose the full entry in single quotes like this:

```
idstools:
  sids:
    disabled:
```

**Reply by 8inaryMata1eao** [reply]
That was a great hint I appreciate the direction.  So i got it to work by doing the following.  The terminal kept expanding the literals but i just put it directly into the files below.  I tried the following with no success.  But the rules directly in the files worked and modifed 1024 rules with no problem.  


`so-rule modify add "re:ET TOR Known" '\$HOME_NET' '[$HOME_NET, !$PUBLIC_SERVERS]'
so-rule modify add '"re:ET TOR Known" '\$HOME_NET' '[$HOME_NET, !$PUBLIC_SERVERS]''
so-rule modify add ''re:ET TOR Known' '\$HOME_NET' '[$HOME_NET, !$PUBLIC_SERVERS]''
`

- /opt/so/conf/idstools/etc/modify.conf
- /opt/so/saltstack/local/pillar/minions/X_manager.sls


\# idstools-rulecat - modify.conf

\# Format: <sid> "<from>" "<to>"

\# Example changing the seconds for rule 2019401 to 3600.
\#2019401 "seconds \d+" "seconds 3600"
"re:ET TOR Known" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"

**Reply by 8inaryMata1eao** [reply]
Guess i spoke to soon.  Doing the manual modification blows up salt.  I had to revert the changes.  
So the question is what syntax to we need to put into `so-rule modify add` to allow a rule with "re:".

ex 1: 
INPUT:`so-rule modify add '\'re:ET TOR Known\' "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"'`
OUTPUT:`so-rule modify add '\'re:ET TOR Known\' "\$HOME_NET" "[$HOME_NET, listPUBLIC_SERVERS]"'
\>`

ex2:
INPUT:`so-rule modify add 're:ET TOR Known "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"'`
OUTPUT:`usage: so-rule modify add [-h] [--apply] SID|REGEX SEARCH_TERM REPLACE_TERM
so-rule modify add: error: the following arguments are required: SEARCH_TERM, REPLACE_TERM`

ex3:
INPUT:`so-rule modify add '"re:ET TOR Known" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"'`
OUTPUT:`usage: so-rule modify add [-h] [--apply] SID|REGEX SEARCH_TERM REPLACE_TERM
so-rule modify add: error: the following arguments are required: SEARCH_TERM, REPLACE_TERM`

ex4:
INPUT:`so-rule modify add ""re:ET TOR Known" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]""`
OUTPUT:`so-rule modify add ""re:ET TOR Known" "\$HOME_NET" "[$HOME_NET, '"re:ET TOR Known" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"'PUBLIC_SERVERS]""
usage: so-rule [-h]  ...
so-rule: error: unrecognized arguments: "re:ET TOR Known" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"PUBLIC_SERVERS]`

ex5:
INPUT:`so-rule modify add "\"re:ET TOR Known\" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]""`
OUTPUT:`so-rule modify add "\"re:ET TOR Known\" "\$HOME_NET" "[$HOME_NET, 're:ET\ TOR\ Known "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"'PUBLIC_SERVERS]""
usage: so-rule modify add [-h] [--apply] SID|REGEX SEARCH_TERM REPLACE_TERM
so-rule modify add: error: the following arguments are required: REPLACE_TERM`

ex6:
INPUT:`so-rule modify add '"re:ET TOR Known"' "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"`
OUTPUT:`so-rule modify add '"re:ET TOR Known"' "\$HOME_NET" "[$HOME_NET, 're:ET\ TOR\ Known "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"'PUBLIC_SERVERS]""PUBLIC_SERVERS]"'PUBLIC_SERVERS]"PUBLIC_SERVERS]"
\>`

**Comment by Hammer-Bear** [comment]
if you still want the rule to trigger but exclude certain ips/ranges you're better off using the suppressions. https://docs.securityonion.net/en/2.3/managing-alerts.html#suppressions

**Reply by 8inaryMata1eao** [reply]
Thanks for the recommendation but I dont believe you can use "re:" with suppression rules.  The problem with that is there are 1000+ rules in the ET TOR Known rule set. Yeah I can do a once off but what about new ET TOR Known rules.  It doesnt scale then.

**Comment by 8inaryMata1eao** [comment]
I tried to modify the rules directly like stated in the [modify documentation](https://docs.securityonion.net/en/2.3/managing-alerts.html#modify-the-sid).
"_If you can’t run so-rule, you can modify the configuration manually in the manager pillar file at /opt/so/saltstack/local/pillar/minions/<managername>_<role>.sls (where <role> is manager, managersearch, standalone, or eval depending on the manager type that was chosen during install). In this file, the idstools section has a modify sub-section where you can add your modifications._ "

When running `so-rule modify list` the system reports `Could not open /opt/so/saltstack/local/pillar/minions/mgr-01_manager.sls` or it works because the rule is allowed but then so-rule-update breaks with python errors

ex1: 
- Set : minions/<managername>_<role>.sls to: 're:ET TOR Known' "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"
- Run salt mgr-01_manager state.apply idstools
- Error: Pillar failed to render with the following messages: Rendering SLS 'minions.mgr-01_manager' failed. Please see master log for details.
-- found unknown escape character
- Systemctl status salt-master : salt-master[1168]: [CRITICAL] Pillar render error: Rendering SLS 'mgr-01_manager' failed. Please see master log for details.

ex2:
- Set : minions/<managername>_<role>.sls to: re:ET TOR Known "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"
- Run salt mtfcu-hou-so-mgr-01_manager state.apply idstools *No Error
- Run so-rule modify list * shows rule correctly
- Run so-rule-update * Error
 "2022-07-21 16:12:01,036 - <INFO> - Forcing Suricata version to 6.0.
Traceback (most recent call last):
  File "/usr/local/bin/idstools-rulecat", line 12, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 841, in main
    modify_filters += load_filters(args.modify)
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 375, in load_filters
    filter = ModifyRuleFilter.parse(line)
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 218, in parse
    raise Exception("Bad number of arguments.")
Exception: Bad number of arguments."
- Contents of idstools/etc/modify.conf: re:ET TOR Known "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"

ex3: 
- Set : minions/<managername>_<role>.sls to:"re:ET TOR Known" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"
- Run salt mgr-01_manager state.apply idstools
- Error: Pillar failed to render with the following messages: Rendering SLS 'minions.mgr-01_manager' failed. Please see master log for details. 
-- found unknown escape character
- Systemctl status salt-master : salt-master[1168]: [CRITICAL] Pillar render error: Rendering SLS 'mgr-01_manager' failed. Please see master log for details.



If I add the rule to idstools/etc/modify.conf its works perfectly but then obviously disappears after salt sync occurs. 
modify.conf: 're:ET TOR Known' "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"
Is there a way I can add the rule to modify.conf and stop it from being over written by salt?

**Comment by 8inaryMata1eao** [comment, marked as answer]
I resolved it on my own.  It appears doing regex for 're:ET TOR' or other types does not work when using "so-rule modify".
If you manually add the rule to the minion file for the manager it then works as expected.  Example below.

1. Modify File: /opt/so/saltstack/local/pillar/minions/MANAGER_manager.sls
```
idstools:
  config:
    oinkcode: ''
    ruleset: ETOPEN
    urls: null
  sids:
    disabled:
    modify:
##### ADD HERE #####
    - '"re:ET TOR Known" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"'
    - '"re:ET Threatview" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS]"'
    - '"re:ET POLICY" "\$HOME_NET" "[$HOME_NET, !$PUBLIC_SERVERS,  !$GOOGLE, !MICROSOFT]"'
```
2. salt '*' state.highstate
3. so-rule-update

**Comment by btdewey** [comment]
Good to know.  We have been manually adding exclusions, but haven't done
any large regular expression changes.
